src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/abi/Binding.java

```diff
@@ -324,19 +359,42 @@ public static Dup dup() {
         return new Binding.Builder();
     }
 
+    @Override
+    public boolean equals(Object o) {
```

Comment:
General question - where do we use equals/hash? Just for testing purposes right?

Reply:
Yeah, we use them at least equals for the CallArranger tests, which is comparing arrays of Bindings.